Source/WebCore/editing/EditingStyle.cpp

```diff
@@ -148,18 +148,19 @@ static RefPtr<CSSValue> extractPropertyValue(ComputedStyleExtractor& computedSty
     return computedStyle.propertyValue(propertyID);
 }
 
-template<typename T>
-int identifierForStyleProperty(T& style, CSSPropertyID propertyID)
-{
-    RefPtr<CSSValue> value = extractPropertyValue(style, propertyID);
-    if (propertyID == CSSPropertyFontStyle && is<CSSFontStyleValue>(value) && downcast<CSSFontStyleValue>(value.get())->isItalicOrOblique())
-        return CSSValueItalic;
-    if (!is<CSSPrimitiveValue>(value))
-        return 0;
-    auto& primitiveValue = downcast<CSSPrimitiveValue>(*value);
-    if (propertyID == CSSPropertyFontWeight && primitiveValue.isNumber() && primitiveValue.doubleValue(CSSUnitType::CSS_NUMBER) >= boldThreshold())
-        return CSSValueBold;
-    return primitiveValue.valueID();
+// This synthesizes CSSValueBold and CSSValueItalic when appropriate, and never returns CSSValueOblique.
+template<typename T> CSSValueID identifierForStyleProperty(T& style, CSSPropertyID propertyID)
+{
+    auto value = extractPropertyValue(style, propertyID);
+    if (auto fontStyleValue = dynamicDowncast<CSSFontStyleValue>(value.get()))
+        return fontStyleValue->obliqueAngle().doubleValue(CSSUnitType::CSS_DEG) >= italicThreshold() ? CSSValueItalic : CSSValueNormal;
```

Comment:
Might be worth adding an assert that the property ID is CSSPropertyFontStyle here.